geocode/query_funcs.py

```
@@ -598,318 +597,138 @@ def geonames_geocode_plain_text(in_text,username='demo',iso_2=None):
-    centroid_long = np.mean([max([i['point'].long for i in point_meta]),
-                             min([i['point'].long for i in point_meta])])
-    centroid_arg = '{},{}'.format(np.round(centroid_lat,decimals=5),
-                                  np.round(centroid_long,decimals=5))
-    # Create a list of markers and combine them with the | operator
-    marker_list = []
-    for point_dict in point_meta:
-        marker_list.append('{},{},{}'.format(point_dict['point'].lat,
-                                             point_dict['point'].long,
-                                             point_dict['osm_meta']))
-    markers_arg = '|'.join(marker_list)
-
-    # Format as dict and then translate to PHP query format
-    osm_args_dict = {'show':1,
-                     'center':centroid_arg,
-                     'zoom':zoom,
-                     'size':dimensions_arg,
-                     'maptype':'mapnik',
-                     'markers':markers_arg}
-    osm_args_url = parse.urlencode(osm_args_dict)
-    
-    full_url = 'http://staticmap.openstreetmap.de/staticmap.php?{}'.format(osm_args_url)
-    return full_url
+def get_total_buffer(locs_list):
+    """Take a location or list of locations and get the approximate distance (in
+    km) of the bounding box diagonal."""
+    # If a single location, force to list
+    locs_list = np.atleast1d(locs_list)
+    pass
 
 
-def summary_maps(df,address_col,out_file_path,gmaps_key,project_name="Summary Maps"):
-    """This function takes the output of geocoding and returns summary text
-    and maps for each geocoding result.
-    
-    Inputs
-    df: The dataframe output from geocoding
-    address_col (str): The name of the column containing addresses for geocoding
-    out_file_path: Where the PDF will be saved, including the .PDF extension
-    project_name: Will be printed in the title and at the top of each page
-    """
+class WebInterface(object):
```

I'm curious as to whether our inheritance model here could be improved--I'm not sure if we need to have a build_query method in the parent class if it will be overrode by all child classes.

---

We clarified this in person - a few of the methods will fail loudly if a child method has not been written, while other methods will be handled entirely by the parent class.